### PR TITLE
Set hydra update strategy

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**

Changes proposed in this pull request:

- Update production profile
- Set default strategy values
- Add HPA for hydra
- Introduce hydra-maester sidecar, replacing the post-upgrade sync-job

**Related issue(s)**

### CODE REVIEW

File: docs/security/05-05-production-profile.md [hunk 1]
```diff
@@ -114,4 +114,4 @@ Follow these steps to migrate your Oauth2 server to the production profile:
 >* [Helm overrides for Kyma installation](/root/kyma/#configuration-helm-overrides-for-kyma-installation)
 >* [Top-level charts overrides](/root/kyma/#configuration-helm-overrides-for-kyma-installation-top-level-charts-overrides)
 
->**TIP:** All the client data registered by Hydra Maester is migrated to the new database as a part of the update process. If you notice missing or inconsistent data, delete the Hydra Maester Pod to force reconciliation. For more information, read about [Hydra Maester controller and Oauth2 client registration in Kyma](#details-o-auth2-and-open-id-connect-server).
+>**TIP:** All the client data registered by Hydra Maester is migrated to the new database as a part of the update process. During this process clients will not be available which might result in 401 on issuing token. If you notice missing or inconsistent data, delete the Hydra Maester Pod to force reconciliation. For more information, read about [Hydra Maester controller and Oauth2 client registration in Kyma](#details-o-auth2-and-open-id-connect-server).
```
Comment (bszwarc) [inline docs/security/05-05-production-profile.md:117]:
```suggestion
>**TIP:** All the client data registered by Hydra Maester is migrated to the new database as a part of the update process. During this process, the clients will not be available which may result in 401 on issuing the token. If you notice missing or inconsistent data, delete the Hydra Maester Pod to force reconciliation. For more information, read about [Hydra Maester controller and Oauth2 client registration in Kyma](#details-o-auth2-and-open-id-connect-server).
```

File: resources/ory/files/secret-migration.sh [hunk 2]
```diff
@@ -9,14 +9,13 @@ readonly TARGET_SECRET_NAME='{{ include "ory.fullname" . }}-hydra-credentials'
 readonly LOCAL_SECRETS_DIR="/etc/secrets"
 
 trap "echo 'error' && exit 1" TERM
```
Comment (jakkab) [inline resources/ory/files/secret-migration.sh:11]: please remove, we don't send TERM sig now.

**Pull request merged**